### PR TITLE
HTML-CSS-Popup homework by expero1

### PR DESCRIPTION
**Pull request by expero1**
### HTML-CSS-Popup

[Demo](https://expero1.github.io/html-css-popup/) |
[Code base](https://github.com/expero1/html-css-popup)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/expero1/html-css-popup/index.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	<link rel="stylesheet" href="./style.css">
+	<title>Document</title>
+	
+</head>
+<body>
+	<header>
+		<div class='header-wrapper'>
+			<div class="logo">
+				<img src='img/favicon.png' alt='Kottans'>
+			</div>
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/index.html:17]: The logo should be a link and lead to the home page.

File: submissions/expero1/html-css-popup/index.html [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+	<meta charset="UTF-8">
+	<meta http-equiv="X-UA-Compatible" content="IE=edge">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<link rel="stylesheet" href="./style.css">
+	<title>Document</title>
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/index.html:9]: Add propper page title.

File: submissions/expero1/html-css-popup/index.html [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+			</div>
+			<nav class="main-menu" aria-label="Main">
+				<ul>	
+					<li><a href='https://github.com/expero1/html-css-popup' target="_blank">source code</a></li>
+					<li><a href="#">Mail</a></li>
+					<li><a href="#">Images</a></li>
+				</ul>
+			</nav>
+      <div class="popup-menu-wrapper">
+        <input type="checkbox" id="popup-checkbox" class="popup-checkbox"  />
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/index.html:26]: Use [Prettier](https://prettier.io/) to fix code formatting

File: submissions/expero1/html-css-popup/index.html [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+				<label for="popup-checkbox">
+					<span class="menu-button popup-button"><!-- <img src='img/popup-button.png'>--></span>
+				</label>
+				<nav class="popup-menu" aria-label="Popup navigation">
+					<ul class="popup-menu-inner">
+							<li><a href="#" class="product-btn"><img src="img/calendar.png" alt="Calendar"><span>Calendar</span></a></li>
+							<li><a href="#" class="product-btn"><img src="img/mail.png" alt="Mail" ><span>Mail</span></a></li>
+							<li><a href="#" class="product-btn"><img src="img/news.png" alt="News" ><span>News</span></a></li>
+							<li><a href="#" class="product-btn"><img src="img/maps.png" alt="Maps"><span>Maps</span></a></li>
+							<li><a href="#" class="product-btn"><img src="img/files.png" alt="files"><span>Files</span></a></li>
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/index.html:36]: Read how and when to populate alt attribute value [Как правильно написать alt-текст](https://htmlacademy.ru/blog/articles/alt-text)

File: submissions/expero1/html-css-popup/style.css [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+	-moz-appearance: none;
+	appearance: none;
+	outline: none;
+}
+
+
+.popup-checkbox:checked + label .popup-button,
+.popup-checkbox:focus + label .popup-button {
+	outline: 2px solid grey;
+	/*background: no-repeat center/100% url(./img/apps.png);*/
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/style.css:100]: Avoid commented code in your PRs

File: submissions/expero1/html-css-popup/style.css [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+	right: 50%;
+	text-transform: uppercase;
+	z-index: -1;
+	display: block;
+	text-align: center;
+	transform: translateX(50%);
+	font-size: 50px;
+}
+
+footer {
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/style.css:303]: Avoid styling by HTML tags except if you add base styles. [Reason](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css)

File: submissions/expero1/html-css-popup/style.css [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+		top: 60px;
+		right:-150px;
+	}
+
+	.popup-menu-inner {
+		grid-template-columns: repeat(4, 1fr);
+		grid-template-rows: auto;
+		gap:5px;
+	}
+	.popup-menu-inner li a {
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/style.css:402]: Avoid using too specific CSS selectors. [Read more](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!

File: submissions/expero1/html-css-popup/index.html (hunk 1 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html]: Thank You. That's my fault. I was in a hurry and was not attentive. Fixed.

File: submissions/expero1/html-css-popup/index.html (hunk 2 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html]: Thank You. I have added the correct title. Fixed

File: submissions/expero1/html-css-popup/index.html (hunk 4 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html:38]: @A-Ostrovnyy  I carefully read the recommended article. As I understand it, if I use an image inside a link, I should fill in the alt attribute. It is also recommended not to write long alt attributes. What do you think is the best course of action in this case? Is it worth using the alt attribute and how long should it be in this case? I would be grateful for a hint.

File: submissions/expero1/html-css-popup/style.css (hunk 5 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/style.css]: Thank You. I have removed commented code.

File: submissions/expero1/html-css-popup/index.html (hunk 4 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html:38]: Maybe a better solution is "Open Mail App" and so on?

File: submissions/expero1/html-css-popup/index.html (hunk 3 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html:28]: @A-Ostrovnyy  Thanks. Should I do it right now? I just think that formatting the code right now will cause too many changes to be displayed. Will it be convenient to do a code review? Perhaps you should do it a little later? What do you think?

File: submissions/expero1/html-css-popup/style.css (hunk 6 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/style.css]: Thank you. I have read the article with great attention. Corrected according to new knowledge.

File: submissions/expero1/html-css-popup/style.css (hunk 7 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/style.css]: Thanks. Fixed it.

File: submissions/expero1/html-css-popup/index.html (hunk 4 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html]: After some thought, I think it's best not to use alt text in the case of icons. Am I right?

File: submissions/expero1/html-css-popup/index.html (hunk 3 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html]: Thanks. I used prettier for uniform file formatting.

**Comment by expero1** [comment]
Thanks for your attention to my code. I tried to correct all the comments by carefully re-reading the entire code again, which took some time. Hope it's much better now.

File: submissions/expero1/html-css-popup/style.css [hunk 8]
```diff
@@ -0,0 +1,397 @@
...
+  background-color: gray;
+}
+
+.notification-menu {
+  grid-area: notification-menu;
+}
+.user-menu {
+  grid-area: user-menu;
+}
+.user-menu img,
```
Comment (A-Ostrovnyy) [inline submissions/expero1/html-css-popup/style.css:266]: Avoid styling by HTML tags except if you add base styles. [Reason](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css)
Also, avoid using too specific CSS selectors. Read more about [CSS specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good progress!

File: submissions/expero1/html-css-popup/style.css (hunk 8 above)
Comment (expero1) [reply submissions/expero1/html-css-popup/style.css]: Thank you. I've added classes for elements on the page and added class selectors to the CSS file for almost all selectors where I've used the HTML tag before.

File: submissions/expero1/html-css-popup/index.html [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+            </ul>
+            <input
+              type="checkbox"
+              id="show-more-checkbox"
+              class="show-more-checkbox"
+            />
+            <label for="show-more-checkbox" class="btn show-more-btn"
+              >show more</label
+            >
+            <div class="hor-line"></div>
```
[the remark this answers is not in the document]
Comment (expero1) [reply submissions/expero1/html-css-popup/index.html]: Thank you for your comment. Fixed.

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**